components/solr-access-service/ui/src/main/resources/PhenoTips/DiffDiagnosisService.xml

```diff
@@ -55,29 +59,46 @@
       #set ($discard = $ndata.add($i))
     #end
     #if ("$!{request.format}" == 'html')
-#set ($results = $_service.getDifferentialPhenotypes($data, $ndata, $limit))
-#if ($results.size() &gt; 0)
-{{html clean="false"  wiki="false"}}##
-&lt;ul&gt;
-#foreach($item in $results)
-  #if ("$!item.id"!= '' &amp;&amp; "$!item.name" != '')
-  &lt;li class="phenotype"&gt;
-    &lt;span class="id" title="${item.id}"&gt;[${item.id}]&lt;/span&gt;
-    &lt;span class="title"&gt;${item.name}&lt;/span&gt;
-    &lt;span class="hidden term-category"&gt;
-    #foreach ($c in $services.vocabularies.resolveTerm($item.id).get('term_category'))
-      &lt;input type="hidden" value="$c"&gt;
-    #end
-    &lt;/span&gt;
-  &lt;/li&gt;
-  #end
-#end## foreach item
-&lt;/ul&gt;
-{{/html}}
-   #set ($hasOutput = true)
-#else
-   #set ($hasOutput = false)
-#end## results.size() &gt; 0
+      #set ($resultsToFetch = $limit * $page)
+      #set ($results = $_service.getDifferentialPhenotypes($data, $ndata, $resultsToFetch))
+      #if ($results.size() &gt; 0)
+        ##paginate results
+        #set ($from = ($page - 1) * $limit)
```

Comment:
It may be possible that less than the number of requested pages are returned, in which case `subList` will throw an exception.